src/OpenTelemetry/CHANGELOG.md

```diff
@@ -2,6 +2,12 @@
 
 ## Unreleased
 
+* **Breaking Change** Removed support for ParseStateValues if the state
+  implements neither IReadOnlyList nor IEnumberable when `ParseStateValues` is
```

Comment:
```suggestion
  implements neither `IReadOnlyList<KeyValuePair<string, object?>>` nor `IEnumerable<KeyValuePair<string, object?>>` when `ParseStateValues` is
```

Reply:
Same for:
- https://github.com/open-telemetry/opentelemetry-dotnet/pull/4560/files#diff-f1b040b01138ef7a10c01209ca53e58d20b78bb4580eafa0f51030b01d529145R316
- https://github.com/open-telemetry/opentelemetry-dotnet/pull/4560/files#diff-48886d84cf3065379188763f0948382a1b89bf9ebf9bdef6f8146fa6acf4776eR196

Reply:
@CodeBlanch Should we be using this instead?

`IReadOnlyList<out T>` nor `IEnumerable<out T>` where `T` is `KeyValuePair<string, object>`.